test/test_ops.py

```diff
@@ -312,7 +312,7 @@ def test_div(self):
     helper_test_op([(), ()], lambda x,y: x/y)
     helper_test_op(None, lambda x,y: x/y, Tensor.div, forward_only=True, vals=np.array([[5],[1]], dtype=np.int32))
   def test_div_int(self):
-    helper_test_op(None, lambda x: (x/2).to(torch.int), lambda x: x/2, forward_only=True, vals=np.array([[3]], dtype=np.int32))
+    helper_test_op(None, lambda x: x/2, lambda x: x/2, forward_only=True, vals=np.array([[3]], dtype=np.int32))
```

Comment:
checkout #1644
reverting the change in #1644 also makes this test pass
maybe it's relevant? i dunno :D
me noob

Reply:
https://github.com/tinygrad/tinygrad/blob/640e5c36ad839c38159bccd23a23ba5b4c2d8efb/tinygrad/tensor.py#L818

Reply:
Thanks for that, looks promising, specifically the conditions in the line below need modifying as it also allows division of `None` currently.

https://github.com/tinygrad/tinygrad/blob/640e5c36ad839c38159bccd23a23ba5b4c2d8efb/tinygrad/tensor.py#L818

Revisiting #1644 to modify the way `maximum()` works for int tensors would negate the need for `dtypes.is_float(self.dtype)` and potentially the need for any casting of int arguments passed to `div`.

Is there any reason why `(self + x) / 2` is used:

https://github.com/tinygrad/tinygrad/blob/640e5c36ad839c38159bccd23a23ba5b4c2d8efb/tinygrad/tensor.py#L841

Surely at the point it is confirmed that neither `self < x` or `self > x` the fill value can be either `self` or `x`?

Reply:
> Is there any reason why (self + x) / 2 is used


I think it's to get the backwards working?

Reply:
ok it looks like `(self + x) / 2` is used to treat cases where self == x, for `forwards` and `backwards`